packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -418,50 +418,76 @@ class ThemeData {
+      primaryColorBrightness: primaryColorBrightness ?? this.primaryColorBrightness,
+      accentColor: accentColor ?? this.accentColor,
+      accentColorBrightness: accentColorBrightness ?? this.accentColorBrightness,
+      canvasColor: canvasColor ?? this.canvasColor,
+      scaffoldBackgroundColor: scaffoldBackgroundColor ?? this.scaffoldBackgroundColor,
+      cardColor: cardColor ?? this.cardColor,
+      dividerColor: dividerColor ?? this.dividerColor,
+      highlightColor: highlightColor ?? this.highlightColor,
+      splashColor: splashColor ?? this.splashColor,
+      selectedRowColor: selectedRowColor ?? this.selectedRowColor,
+      unselectedWidgetColor: unselectedWidgetColor ?? this.unselectedWidgetColor,
+      disabledColor: disabledColor ?? this.disabledColor,
+      buttonColor: buttonColor ?? this.buttonColor,
+      secondaryHeaderColor: secondaryHeaderColor ?? this.secondaryHeaderColor,
+      textSelectionColor: textSelectionColor ?? this.textSelectionColor,
+      textSelectionHandleColor: textSelectionHandleColor ?? this.textSelectionHandleColor,
+      backgroundColor: backgroundColor ?? this.backgroundColor,
+      dialogBackgroundColor: dialogBackgroundColor ?? this.dialogBackgroundColor,
+      indicatorColor: indicatorColor ?? this.indicatorColor,
+      hintColor: hintColor ?? this.hintColor,
+      errorColor: errorColor ?? this.errorColor,
+      textTheme: textTheme ?? this.textTheme,
+      primaryTextTheme: primaryTextTheme ?? this.primaryTextTheme,
+      accentTextTheme: accentTextTheme ?? this.accentTextTheme,
+      iconTheme: iconTheme ?? this.iconTheme,
+      primaryIconTheme: primaryIconTheme ?? this.primaryIconTheme,
+      accentIconTheme: accentIconTheme ?? this.accentIconTheme,
+      platform: platform ?? this.platform,
     );
   }
 
+  static ThemeData _lastBaseTheme;
+  static TextTheme _lastLocalTextGeometry;
+  static ThemeData _memoizedLocalizeResult;
+
   /// Returns a new theme built by merging [baseTheme] into the text geometry
   /// provided by the [localTextGeometry].
   ///
   /// The [TextStyle.inherit] field in the text styles provided by
   /// [localTextGeometry] must be set to `true`.
   static ThemeData localize(ThemeData baseTheme, TextTheme localTextGeometry) {
+    // WARNING: this method memoizes the result based on the previously seen
+    // baseTheme and localTextGeometry. Memoization is safe because all inputs
+    // and outputs of this function are deeply immutable, and the computations
+    // are referentially transparent. It only short-circuits the computation if
+    // the new inputs are identical() to the previous ones. It does not use the
+    // == operator, which performs a costly deep comparison. This memoization
+    // saves us ~6% on the stock_build_iteration benchmark.
+    //
+    // When changing this method, make sure the memoization logic is correct. Do
+    // not forget about the two hardest things in computer science:
```

Comment:
to avoid broken links, i recommend inlining the advice you want to reference here.

Reply:
Done.